metaflow/plugins/tagging_cli.py

```diff
@@ -0,0 +1,424 @@
+        #  - the comma separated list of system tags
+        #  - the comma separated list of user tags
+        # Columns are separated by a semicolon surrounded by a space on each side
+        # to make it visually clear when there are no system tags for example
+        obj.echo_always(' ; '.join([
+            cat,
+            ','.join(sorted(system_tags)),
+            ','.join(sorted(all_tags.difference(system_tags)))]))
+
+
+def tag_category_format(obj, cat, runs, is_system):
+    if not obj.is_quiet:
+        # Pretty printing computations
+        max_length = max([len(t) for t in runs] if runs else [0])
+
+        all_runs = sorted(runs)
+
+        num_runs = len(all_runs)
+
+        # We consider 4 spaces in between column and we consider 120 characters total
+        # width and we want 120 >= column_count*max_length + (column_count - 1)*4
+        column_count = 124 // (max_length + 4)
+        if column_count == 0:
+            # Make sure we have at least 1 column even for very very long tags
+            column_count = 1
+            words_per_column = num_runs
+        else:
+            words_per_column = (num_runs + column_count - 1) // column_count
+
+        # Extend all_runs by empty runs to be able to easily fill up the lines
+        addl_runs = words_per_column * column_count - num_runs
+        if addl_runs > 0:
+            all_runs.extend([' ' for _ in range(addl_runs)])
+
+        lines = []
+        lines.append('For tag %s' % (cat if is_system else '*%s*' % cat))
+        for i in range(words_per_column):
+            line_values = [all_runs[i+j*words_per_column] for j in range(column_count)]
+            length_values = [max_length for l in line_values]
+            formatter = "    ".join(["{:<%d}" % l for l in length_values])
+            lines.append(formatter.format(*line_values))
+        lines.append("")
+        obj.echo('\n'.join(lines), err=False)
+    else:
+        # In quiet mode, we display things much more simply so it is parseable.
+        # When displaying by tag, we output twp columns:
+        #  - the tag (user:<tag> or sys:<tag>)
+        #  - the comma separated list of runs
+        # Columns are separated by a semicolon surrounded by a space on each side
+        # to make it consistent with the by_run listing
+        obj.echo_always(' ; '.join([
+            'system:%s' % cat if is_system else 'user:%s' % cat,
+            ','.join(runs)]))
+
+
+def print_tags(obj, metaflow_run_client):
+    system_tags = {metaflow_run_client.pathspec: metaflow_run_client.system_tags}
+    all_tags = {metaflow_run_client.pathspec: metaflow_run_client.tags}
+    return format_tags(obj, system_tags, all_tags, by_tag=False)
+
+
+def is_prod_token(token):
+    return token is not None and token.startswith("production:")
+
+
+def obtain_metaflow_run_client(obj, run_id, user_namespace):
+    if is_prod_token(user_namespace):
```

Comment:
I am not sure this check is correct. I can pass in a different namespace for a scheduled run. Also, can we consider allowing modifying a scheduled run's tags?

Reply:
what would a better check be?

Reply:
A better check would be to ensure you are in the right namespace.